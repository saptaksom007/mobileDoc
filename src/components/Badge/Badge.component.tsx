import * as React from 'react'

import { View, ViewStyle, TextStyle } from 'react-native'
import { Small } from 'components/CustomText/CustomText.component'
import styles from './Badge.styles'

interface Props {
  value?: string | number
  size?: number
  color?: string
  fontSize?: number
}

export const Badge = (props: Props) => {
  const { value, size, color, fontSize } = props
  // Empty badge =>
  if (!value || (typeof value === 'number' && value < 0)) {
    return (
      <View
        style={
          [
            styles.emptyBadge,
            size && { width: size, height: size },
          ] as ViewStyle
        }
      />
    )
  }
  return (
    <View
      style={
        [
          styles.badge,
          size && { width: size, height: size, borderRadius: size / 2 },
          color && { backgroundColor: color },
        ] as ViewStyle
      }
    >
      <Small style={[styles.badgeText, fontSize && { fontSize }] as TextStyle}>
        {String(value)}
      </Small>
    </View>
  )
}
