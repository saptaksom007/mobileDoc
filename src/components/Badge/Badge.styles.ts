import { StyleSheet, Platform } from 'react-native'
import normalize from 'react-native-elements/src/helpers/normalizeText'

export default StyleSheet.create({
  badge: {
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: '#e74c3c',
    borderRadius: normalize(10),
    width: normalize(20),
    height: normalize(20),
    zIndex: 1000
  },
  badgeText: {
    textAlign: 'center',
    color: '#fff',
    backgroundColor: 'transparent',
    width: normalize(20),
    height: normalize(20),
    borderRadius: normalize(10),
    lineHeight: Math.round(normalize(20)),
    ...Platform.select({
      android: {
        marginLeft: -1,
        fontSize: normalize(11)
      },
      ios: {
        fontSize: normalize(12)
      }
    })
  },
  emptyBadge: {
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: 'transparent',
    width: normalize(20),
    height: normalize(20)
  }
})
