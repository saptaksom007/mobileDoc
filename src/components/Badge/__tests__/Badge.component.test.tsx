import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'
import { Badge } from 'components/Badge/Badge.component'

it('<Badge/> render correctly', () => {
  const tree = renderer.create(<Badge value={1} size={10} />).toJSON()
  expect(tree).toMatchSnapshot()
})
