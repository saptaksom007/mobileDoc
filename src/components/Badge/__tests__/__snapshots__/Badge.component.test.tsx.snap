// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Badge/> render correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#e74c3c",
        "borderRadius": 12.5,
        "height": 25,
        "justifyContent": "center",
        "width": 25,
        "zIndex": 1000,
      },
      Object {
        "borderRadius": 5,
        "height": 10,
        "width": 10,
      },
      undefined,
    ]
  }
>
  <Text
    allowFontScaling={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontFamily": "TitilliumWeb-Regular",
          "fontSize": 17.5,
          "lineHeight": 22,
          "overflow": "visible",
        },
        Object {
          "color": "rgba(27, 106, 158, 0.85)",
          "fontSize": 15,
          "lineHeight": 18,
        },
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderRadius": 12.5,
            "color": "#fff",
            "fontSize": 15,
            "height": 25,
            "lineHeight": 25,
            "textAlign": "center",
            "width": 25,
          },
          undefined,
        ],
      ]
    }
  >
    1
  </Text>
</View>
`;
